__tests__/server/utils/onModuleLoad.spec.jsx

```diff
@@ -253,6 +253,20 @@ describe('onModuleLoad', () => {
     expect(() => onModuleLoad({ module: { [CONFIGURATION_KEY]: configuration, [META_DATA_KEY]: { version: '1.0.11' } }, moduleName: 'my-awesome-module' })).toThrowErrorMatchingSnapshot();
   });
 
+  it('logs a warning if the root module provides an incompatible version of a required external and DANGEROUSLY_ACCEPT_BREAKING_EXTERNALS is set to true', () => {
+    process.env.DANGEROUSLY_ACCEPT_BREAKING_EXTERNALS = true;
+    RootModule[CONFIGURATION_KEY].providedExternals = {
+      'dep-a': { version: '2.1.0', module: () => 0 },
+    };
+    const configuration = {
+      requiredExternals: {
+        'dep-a': '^2.1.1',
+      },
+    };
+    expect(() => onModuleLoad({ module: { [CONFIGURATION_KEY]: configuration, [META_DATA_KEY]: { version: '1.0.10' } }, moduleName: 'my-awesome-module' })).not.toThrow();
+    expect(consoleWarnSpy).toHaveBeenCalledTimes(1);
+  });
+
```

Comment:
maybe also include test that checks if `DANGEROUSLY_ACCEPT_BREAKING_EXTERNALS` is unset that it throws the error

Reply:
That is already tested on this line
https://github.com/americanexpress/one-app/blob/main/__tests__/server/utils/onModuleLoad.spec.jsx#L225